src/Analyzers/CSharp/Tests/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs

```diff
@@ -1966,5 +1966,46 @@ public async Task TopLevelStatement_FollowedWithThrow()
 
             await test.RunAsync();
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertSwitchStatementToExpression)]
+        [WorkItem(48006, "https://github.com/dotnet/roslyn/issues/48006")]
+        public async Task TestOnMultiCaseSection_String_CSharp9()
+        {
+            var testCode = @"
+class Program
+{
+    bool M(string s)
+    {
+        [|switch|] (s)
+        {
+	        case ""Last"":
+            case ""First"":
+            case ""Count"":
+                return true;
+            default:
+                return false;
+        }
+    }
+}";
+            var fixedCode = @"
+class Program
+{
+    bool M(string s)
+    {
+        return s switch
+        {
+            ""Last"" or ""First"" or ""Count"" => true,
+            _ => false,
+        };
+    }
+}";
+
+            await new VerifyCS.Test
+            {
+                TestCode = testCode,
+                FixedCode = fixedCode,
+                LanguageVersion = CSharp9,
```

Comment:
Does this need to be tested on earlier version?

Reply:
i'm not sure what you mean.  can you clarify?

Reply:
I meant the exact same test for C# 8 or any version before C# 9.

Reply:
no.  'or' patterns aren't available in C# 8.

Reply:
@CyrusNajmabadi Yes I know. I meant to confirm that the scenario in #48006 doesn't happen with previous version. (i.e. the test should probably check that no code fix is offerred at all for C# 8)

Reply:
I'm satisfied this is covered with existing tests (like TestMissingOnMultiCaseSection).